### DISCUSSION TITLE
Disabling/Removing `Symbol(pinia)` warnings in test console?

### DISCUSSION THREAD

**Discussion by Sensanaty** [vuejs/pinia#2635] [Help and Questions] [answered]
I've got pinia setup with Vitest, and it all works fine but I do have 1 annoyance, and it's that my test logs are *filled* with this message:

`App already provides property with key "Symbol(pinia)". It will be overwritten with the new value.`

I suspect this is because I have `createTestingPinia` in my global `setup.ts` file (which houses all the global test-related stuff like `config.global.components` and stuff like that), but I also override things here in some specific tests.

My setup file has this:

```ts
config.global.plugins = [
  createTestingPinia({ createSpy: vi.fn }),
];
```

And then, in component tests where a specific store is involved, or if I need to change Pinia's global settings (like disabling `action` mocking when I want to actually test calling a store action and its side effects) I override these global options with something like this

```ts
mount(MyComponent, { 
  global: {
    plugins: [ createTestingPinia({ initialState: {} }) ]
  }
})
```

It works fine, but for every single test where I do this I get that above message. Is there any way of disabling that, because it doesn't seem like it's relevant for me?

**Comment by posva** [comment, marked as answer]
The warning comes from Vue, not pinia. You should override the global objects (if that's the way you are doing it) when passing options to `mount()`. You can clear the plugins array `config.global.plugins = []`.

However, I recommend you to **remove the `createTestingPinia()` from the global plugin settings**. Only add it in tests where it's needed.
